roles/scenario13/templates/create_env.sh.j2

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+set -e
+
+{{ osp_get_credentials_cmd }}
+
+if ! {{ oc_osp_cmd }} openstack network show {{ public_network }} > /dev/null 2> /dev/null; then
+    {{ oc_osp_cmd }} openstack network create --external --provider-network-type flat --provider-physical-network datacentre {{ public_network }}
+    {{ oc_osp_cmd }} openstack subnet create --no-dhcp --gateway 10.0.0.1 --network {{ public_network }} --subnet-range 10.0.0.0/24 --allocation-pool start=10.0.0.151,end=10.0.0.254 {{ public_network }}_subnet
+fi
+
+if ! {{ oc_osp_cmd }} openstack image show {{ image_name }} > /dev/null 2> /dev/null; then
+    # Create Cirros image
+    if [ ! -f {{ image_file_name }} ]; then
+        curl -L {{ image_url }} > {{ image_file_name }}
+    fi
+    {{ oc_osp_cmd }} openstack image create {{ image_name }} --disk-format qcow2 --container-format bare --public < {{ image_file_name }}
+fi
+
+{{ oc_osp_cmd }} openstack network create --provider-network-type vlan --provider-segment {{ project_net_segment }} --provider-physical-network {{ project_net_physnet }} scenario-{{ scenario }}-network
+
+{{ oc_osp_cmd }} openstack subnet create --network scenario-{{ scenario }}-network scenario-{{ scenario }}-subnet --subnet-range 192.168.1{{ scenario }}.0/24
+
+{{ oc_osp_cmd }} openstack router create scenario-{{ scenario }}-router
+
+{{ oc_osp_cmd }} openstack router set --enable-snat --external-gateway {{ public_network }} scenario-{{ scenario }}-router
+{{ oc_osp_cmd }} openstack router add subnet scenario-{{ scenario }}-router scenario-{{ scenario }}-subnet
+
+{{ oc_osp_cmd }} openstack flavor show m1.small > /dev/null 2> /dev/null || {{ oc_osp_cmd }} openstack flavor create m1.small --disk 1 --vcpus 1 --ram 512
+
+{{ oc_osp_cmd }} openstack security group create scenario-{{ scenario }}-sg
+{{ oc_osp_cmd }} openstack security group rule create --ingress --protocol icmp scenario-{{ scenario }}-sg
+{{ oc_osp_cmd }} openstack security group rule create --ingress --protocol tcp --dst-port 22 scenario-{{ scenario }}-sg
+{{ oc_osp_cmd }} openstack security group rule create --ingress --protocol tcp --dst-port {{ pf_tcp_port }} scenario-{{ scenario }}-sg
+
+# TODO(slaweq): We can revert that to use OSC to generate ssh keys for us once
+#   we will be using python-openstackclient >= 6.3.0 which contains patch
+#   https://review.opendev.org/c/openstack/python-openstackclient/+/881822
+if type ssh-keygen > /dev/null 2>&1; then
```

Comment:
I don't understand this condition. ssh-keygen is not installed at some boxes?

Reply:
actually this is not needed, thx